Mention lowercase doctype in style guide

Ipshita29 opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description
<!-- ✍️ Summarize your changes in one or two sentences -->
Updated the <!doctype html> declaration to <!DOCTYPE html> in the HTML Code Style Guide to align with standard conventions and best practices.

### Motivation
<!-- ❓ Why are you making these changes and how do they help readers? -->
The uppercase <!DOCTYPE html> is the widely accepted format for declaring the document type in HTML5. Ensuring consistency helps avoid confusion among readers and follows best coding practices.

<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->

Ipshita29 committed:
Update index.md

Corrected doctype declaration to uppercase in HTML Code Style Guide.

dipikabh commented:
Hello @Ipshita29, welcome to MDN 🎉 Thank you for your contribution and attention to detail!

Doctype declaration is case-insensitive and on MDN, we have standardized on using the lowercase version as our style convention. This is noted in the glossary entry for [Doctype](https://developer.mozilla.org/en-US/docs/Glossary/Doctype): "The doctype is case-insensitive. The convention of MDN code examples is to use lowercase, but it's also common to write it as `<!DOCTYPE html>`." 

Instead of changing the casing, would you be interested in adding a note about the case-insensitivity of this declaration, similar to what we have on the glossary page?

Ipshita29 committed:
Added a note about case insensitivity of doctype

Reverted any casing changes to the doctype declaration as per MDN guidelines.
Added a note clarifying that the doctype is case-insensitive while MDN follows the lowercase convention.
Ensured the note follows MDN’s formatting standards.

Ipshita29 commented:
Thank you, @dipikabh, for this opportunity. I have made the requested changes. Could you kindly review them?

Josh-Cena commented on files/en-us/mdn/writing_guidelines/writing_style_guide/code_style_guide/html/index.md:
````diff
@@ -29,6 +29,8 @@ You should use the HTML5 doctype. It is short, easy to remember, and backwards c
 <!doctype html>
 ```
+> [!NOTE]  
+> The doctype declaration (`<!doctype html>`) is case-insensitive. While MDN follows the convention of using lowercase, it is also commonly written as `<!DOCTYPE html>`.  
````
This is our style guide. You shouldn't need to say what else is okay; just say what is okay on MDN.

```suggestion
MDN always writes the doctype declaration in all-lowercase: `<!doctype html>`. This is consistent with our general convention to use lowercase where possible, such as HTML tags, attributes, or hexadecimal digits.
```

dipikabh commented on files/en-us/mdn/writing_guidelines/writing_style_guide/code_style_guide/html/index.md:
````diff
@@ -29,6 +29,8 @@ You should use the HTML5 doctype. It is short, easy to remember, and backwards c
 ```html example-good
 <!doctype html>
 ```
+> [!NOTE]  
+> The doctype declaration (`<!doctype html>`) is case-insensitive. While MDN follows the convention of using lowercase, it is also commonly written as `<!DOCTYPE html>`.  
````
Thanks @Josh-Cena for chiming in. Definitely, we need to emphasize the convention we use on MDN - so the second sentence from @Ipshita29 needs a little tweaking. But also, we do first need to state that the declaration is case-insensitive. This can be followed by the extra context you've provided for this convention.

I propose the following:

```suggestion
The doctype declaration is case-insensitive. On MDN, we follow the convention of writing it in lowercase. This is consistent with our general convention to use lowercase in case-insensitive contexts, such as HTML tags, attributes, and hexadecimal values.
```

Josh-Cena replied:
IMO, we don't need to state facts and we should assume readers already know everything about the subject matter; no strong preferences though.

Ipshita29 replied:
Sure I will work on the following changes.

Ipshita29 replied:
I've made the requested changes.Please review and let me know if any further adjustments are needed. Thank you!

Ipshita29 committed:
Update index.md – Improve casing guidelines

Renamed the "Case" section to "Casing convention on MDN" for better clarity.
Updated the section to emphasize using lowercase for HTML tags, including doctype, element names, and attributes.
Added a glossary link to the "Doctype" section for easy reference.

Josh-Cena commented on files/en-us/mdn/writing_guidelines/writing_style_guide/code_style_guide/html/index.md:
````diff
@@ -24,11 +24,13 @@ Prettier formats all the code and keeps the style consistent. Nevertheless, ther
 <!doctype html>
 ```
+> [!NOTE]  
+> The <!doctype html> declaration is case-insensitive and is commonly written as <!DOCTYPE html>.
````
Again don't say this. It doesn't matter in the style guide how it's "commonly written". What matters is what should be written on MDN. This is a prescriptive document, not descriptive.

dipikabh replied:
```suggestion
```

Josh-Cena commented on files/en-us/mdn/writing_guidelines/writing_style_guide/code_style_guide/html/index.md:
```diff
@@ -90,9 +92,9 @@ This is perfectly understandable and works fine. If a boolean HTML attribute is
+## Casing convention on MDN
 
-Use lowercase for all element names and attribute names/values because it looks neater and means you can write markup faster. For example:
+Use lowercase for all HTML tags, including the doctype declaration, element names, and attribute names/values, which are case-insensitive. This creates a consistent appearance and allows for faster markup writing. This convention applies to all case-insensitive contexts.
```
```suggestion
Use lowercase for all case-insensitive constructs, including the doctype declaration, element names, and attribute names/values. This creates a consistent appearance and allows for faster markup writing.
```

dipikabh committed:
Apply suggestions from code review

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

dipikabh commented on files/en-us/mdn/writing_guidelines/writing_style_guide/code_style_guide/html/index.md:
````diff
@@ -24,11 +24,13 @@ Prettier formats all the code and keeps the style consistent. Nevertheless, ther
 ```html example-good
 <!doctype html>
 ```
+> [!NOTE]  
````
Agree, no need to add this note here. The lower-case convention on MDN applies to constructs other than doctype and the "Casing" section explains that convention

dipikabh replied:
```suggestion
```

dipikabh committed:
Apply suggestions from code review

dipikabh reviewed (approved):
Thanks for your contribution!
